sdv/multi_table/base.py

```diff
@@ -162,6 +163,29 @@ def set_address_columns(self, table_name, column_names, anonymization_level='ful
         self._validate_table_name(table_name)
         self._table_synthesizers[table_name].set_address_columns(column_names, anonymization_level)
 
+    def _detect_single_table_cag(self, patterns):
+        """Filter out single table CAG patterns from the list of patterns.
+
+        Args:
+            patterns (list):
+                A list of CAG patterns to filter.
+        """
+        idx_single_table = None
+        seen_single_table = False
+        for idx, pattern in enumerate(patterns):
+            if self._has_seen_single_table_constraint and pattern._is_single_table is False:
+                raise SynthesizerInputError(
+                    'Cannot apply multi-table constraint after single-table constraint has '
+                    'been applied.'
+                )
+
+            if pattern._is_single_table and not seen_single_table:
+                self._has_seen_single_table_constraint = True
+                seen_single_table = True
```

Comment:
why do we need both of these?

Reply:
I updated it in [69e0b63](https://github.com/sdv-dev/SDV/pull/2531/commits/69e0b6356458e286a1aa5385e87d8cc399df2567) so we keep the class attribute only.